kratos/python_scripts/assign_mpcs_to_neighbours_process.py

```diff
@@ -0,0 +1,178 @@
+        self.computing_model_part_name = settings["computing_model_part_name"].GetString() #mpcs are applied to computing model part 
+        
+        # Get the slave model part 
+        if not settings["slave_model_part_name"].GetString():
+            raise Exception("\'slave_model_part_name\' not provided. Please specify the slave model part.")
+        self.slave_model_part_name = settings["slave_model_part_name"].GetString()
+        
+        # Get the master model part 
+        if not settings["master_model_part_name"].GetString():
+            raise Exception("\'master_model_part_name\' not provided. Please specify the master model part.")
+        self.master_model_part_name = settings["master_model_part_name"].GetString()
+
+        # Search radius for the mpcs
+        self.search_radius = settings["search_radius"].GetDouble()
+
+        # Minimum number of neighboring nodes retrieved from the search radius
+        self.minimum_number_of_neighbouring_nodes = settings["minimum_number_of_neighbouring_nodes"].GetInt()
+
+        # Apply mpcs at each time step (True) or only once (False)
+        self.assign_mpcs_every_time_step = settings["assign_mpcs_every_time_step"].GetBool()
+
+        # Retrieve and check if variables exist
+        variable_names = settings["variable_names"].GetStringArray()
+        if len(variable_names) == 0:
+            err_msg = "The variable names need to be specified by the user in the \'variable_names\' string array."
+            raise Exception(err_msg)
+        if any(variable_names.count(var_name) > 1 for var_name in variable_names):
+            err_msg = "There are repeated variables in the \'variable_names\' string array."
+            raise Exception(err_msg)
+        variable_names.sort()
+
+        self.variables_list = []
+        for var_name in variable_names:
+            if not KM.KratosGlobals.HasVariable(var_name):
+                err_msg = "\'{}\' variable in \'variable_names\' is not in KratosGlobals. Please check provided value.".format(var_name)
+            if not KM.KratosGlobals.GetVariableType(var_name):
+                err_msg = "\'{}\' variable in \'variable_names\' is not double type. Please check provide double type variables (e.g. [\"DISPLACEMENT_X\",\"DISPLACEMENT_Y\"]).".format(var_name)
+            self.variables_list.append(KM.KratosGlobals.GetVariable(var_name))
+        debug = True
+
+    def ExecuteInitialize(self):
+        """ This method is executed at the begining to initialize the process
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        ## Assign master and slave model parts
+        self.computing_model_part = self.model.GetModelPart(self.computing_model_part_name)
+        self.slave_model_part = self.model.GetModelPart(self.slave_model_part_name)
+        self.master_model_part = self.model.GetModelPart(self.master_model_part_name)
```

Comment:
You have member variables for both model and model parts. Either one or the other. In this case it might be easier just to move these three lines to the constructor and avoid saving the model.

If you're doing this because you cannot assume that at construction time of this process the model will be filled, please put a comment explaining so, as this is not common for processes.